src/sage/combinat/designs/designs_pyx.pyx

```diff
@@ -15,6 +15,203 @@ from cysignals.memory cimport sig_malloc, sig_calloc, sig_realloc, sig_free
+        ....:      (0, 0, 1, 1, 1, 0, 2, 1, 1, 2, 1, 0, 1),
+        ....:      (2, 0, 0, 1, 1, 1, 0, 2, 1, 1, 2, 1, 0),
+        ....:      (0, 2, 0, 0, 1, 1, 1, 0, 2, 1, 1, 2, 1),
+        ....:      (2, 0, 2, 0, 0, 1, 1, 1, 0, 2, 1, 1, 2),
+        ....:      (1, 2, 0, 2, 0, 0, 1, 1, 1, 0, 2, 1, 1),
+        ....:      (2, 1, 2, 0, 2, 0, 0, 1, 1, 1, 0, 2, 1),
+        ....:      (2, 2, 1, 2, 0, 2, 0, 0, 1, 1, 1, 0, 2),
+        ....:      (1, 2, 2, 1, 2, 0, 2, 0, 0, 1, 1, 1, 0),
+        ....:      (0, 1, 2, 2, 1, 2, 0, 2, 0, 0, 1, 1, 1),
+        ....:      (2, 0, 1, 2, 2, 1, 2, 0, 2, 0, 0, 1, 1),
+        ....:      (2, 2, 0, 1, 2, 2, 1, 2, 0, 2, 0, 0, 1),
+        ....:      (2, 2, 2, 0, 1, 2, 2, 1, 2, 0, 2, 0, 0),
+        ....:      (0, 2, 2, 2, 0, 1, 2, 2, 1, 2, 0, 2, 0),
+        ....:      (0, 0, 2, 2, 2, 0, 1, 2, 2, 1, 2, 0, 2),
+        ....:      (1, 0, 0, 2, 2, 2, 0, 1, 2, 2, 1, 2, 0),
+        ....:      (0, 1, 0, 0, 2, 2, 2, 0, 1, 2, 2, 1, 2),
+        ....:      (1, 0, 1, 0, 0, 2, 2, 2, 0, 1, 2, 2, 1),
+        ....:      (2, 1, 0, 1, 0, 0, 2, 2, 2, 0, 1, 2, 2),
+        ....:      (1, 2, 1, 0, 1, 0, 0, 2, 2, 2, 0, 1, 2),
+        ....:      (1, 1, 2, 1, 0, 1, 0, 0, 2, 2, 2, 0, 1),
+        ....:      (2, 1, 1, 2, 1, 0, 1, 0, 0, 2, 2, 2, 0),
+        ....:      (0, 2, 1, 1, 2, 1, 0, 1, 0, 0, 2, 2, 2),
+        ....:      (1, 0, 2, 1, 1, 2, 1, 0, 1, 0, 0, 2, 2),
+        ....:      (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0))
+        sage: is_covering_array(C,parameters=True)
+        (True, (53, 3, 13, 3))
+
+    """
+    from itertools import product, combinations
+    from copy import deepcopy
+
+    number_rows=len(array)
+    number_columns=len(array[0])
+
+    for row in range(number_rows):
+        if len(array[row]) != number_columns:
+            raise ValueError("Not all rows are the same length, row {} is not the same length as row 0".format(row))
+
+    if symbol_set is None:
+        symbol_set = list({x for l in array for x in l})
+        symbol_set.sort()
+    else:
+        for l in array:
+            for x in l:
+                if x not in symbol_set:
+                    raise ValueError("{} appears in the array but not in the given symbol set".format(x))
+
+    if strength is None:
+        wstrength = 1
+    else:
+        wstrength = strength
+    finished = False
+
+    # if no strength inputted, tries increasing values for t until one
+    # does not work. If strength is inputted ends after one check
+    while finished is False:
+        # A dictionary to count how many times each t-tuple appears in
+        # the array
+        tupledict={}
+        a=[ttuple for ttuple in product(symbol_set,repeat=wstrength)]
+        for item in a:
+            tupledict.update({item:0})
+
+        # Iterates over every possible selections of t columns, and
+        #count the t-tuples appearing in each selection
+        for comb in combinations(range(number_columns),wstrength):
+            wdict=deepcopy(tupledict)
```

Comment:
I would suggest to not use `deepcopy` but instead use the constructor I suggested in my first comment. Namely
```python
wdict = {item: 0 for item in product(symbol_set, repeat=wstrength)}
```
That way you can simply avoid creating `tupledict` at all.